File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.go
func maxProduct(nums []int) int {

    a:=0
    b:=0

    for i:=0;i<len(nums);i++{

        if nums[i]>a{
            b=a
            a=nums[i]

        }else if nums[i]>b{
            b=nums[i]
        }
    }
    return (a-1)*(b-1)
}

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.cs
public class Solution {
    public int MaxProduct(int[] nums) {

        int a=0;
        int b=0;

        for(int i=0;i<nums.Length;i++){
            if(nums[i]>a){
                b=a;
                a=nums[i];
            }
            else if(nums[i]>b){
                b=nums[i];
            }
        }
        return (a-1)*(b-1);
    }
}